justfile

```diff
@@ -0,0 +1,224 @@
+target-arch := if target =~ "x86_64" { "x64"
+    } else if target =~ "i[56]86" { "x86"
+    } else if target =~ "aarch64" { "arm64"
+    } else if target =~ "armv7" { "arm32"
+    } else { "unknown" }
+target-libc := if target =~ "gnu" { "gnu"
+    } else if target =~ "musl" { "musl"
+    } else { "unknown" }
+
+# build output location
+output-ext := if target-os == "windows" { ".exe" } else { "" }
+output-filename := "cargo-binstall" + output-ext
+output-profile-folder := if for-release != "" { "release" } else { "debug" }
+output-folder := if target != target-host { "target" / target / output-profile-folder
+    } else if env_var_or_default("CARGO_BUILD_TARGET", "") != "" { "target" / target / output-profile-folder
+    } else if cargo-buildstd != "" { "target" / target / output-profile-folder
+    } else { "target" / output-profile-folder }
+output-path := output-folder / output-filename
+
+# which tool to use for compiling
+cargo-bin := if use-cross != "" { "cross" } else { "cargo +nightly" }
+
+# cargo compile options
+cargo-profile := if for-release != "" { "release" } else { "dev" }
+
+
+ci-or-no := if ci != "" { "ci" } else { "noci" }
+
+# In release builds in CI, build the std library ourselves so it uses our
+# compile profile, and optimise panic messages out with immediate abort.
+#
+# explicitly disabled on aarch64-unknown-linux-gnu due to a failing build
+cargo-buildstd := if (cargo-profile / ci-or-no) == "release/ci" {
+    if target == "aarch64-unknown-linux-gnu" { ""
+    } else { " -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort" }
+} else { "" }
+
+# In musl release builds in CI, statically link gcclibs.
+cargo-gcclibs := if (cargo-profile / ci-or-no / target-libc) == "release/ci/musl" {
+    " -C link-arg=-lgcc -C link-arg=-static-libgcc"
+} else { "" }
+
+# disable default features in CI for debug builds, for speed
+cargo-no-default-features := if default-features == "false" { " --no-default-features"
+    } else if default-features == "true" { ""
+    } else if (cargo-profile / ci-or-no) == "dev/ci" { " --no-default-features"
+    } else { "" }
+
+cargo-features := trim_end_match(if override-features != "" { override-features
+    } else if (cargo-profile / ci-or-no) == "dev/ci" { "rustls,fancy-with-backtrace," + extra-features
+    } else if (cargo-profile / ci-or-no / target-libc) == "release/ci/musl" { "rustls,fancy-with-backtrace,zstd-thin," + extra-features
+    } else if (cargo-profile / ci-or-no) == "release/ci" { "rustls,fancy-with-backtrace," + extra-features
+    } else { extra-features
+}, ",")
+
+# it seems we can't split debuginfo for non-buildstd builds
+# errors with: "Found a record with an unknown abbreviation code"
+cargo-split-debuginfo := if cargo-buildstd != "" { " --config='profile.release.split-debuginfo=\"packed\"' --config=profile.release.debug=2" } else { "" }
+
+# for ARM64 Windows, use a patched version of ring
+# this should be unnecessary once ring 0.17 is released
```

Comment:
Can we create a new issue for this just in case we all forget about this?

Reply:
done